Blog/views.py

```diff
@@ -85,20 +105,22 @@ def post(self, request):
             username = request.POST['username']
             password = request.POST['password']
             user = User.objects.get(username=username)
-            if user.check_password(password):
-                if user.is_active:
-                    login(request, user)
-                    return redirect("Blog:homepage")
+            if user is not None:
```

Comment:
It can be combined into `if user and user.check_password(password) and user.is_active:` Depending on and and or operator, the second one will be executed only after the first one and so on. Refer short circuiting http://openbookproject.net/thinkcs/python/english3e/conditionals.html#logical-operators